src/ledger/LedgerManagerImpl.cpp

```diff
@@ -265,16 +278,27 @@ LedgerManagerImpl::ApplyState::getSorobanNetworkConfig() const
 }
 
 SorobanNetworkConfig&
-LedgerManagerImpl::ApplyState::getSorobanNetworkConfigToModify()
+LedgerManagerImpl::ApplyState::getSorobanNetworkConfigToModify(
```

Comment:
`getSorobanNetworkConfig` const function should also enforce APPLYING state, right?

Reply:
I don't think there's a good way to do this since we call `getSorobanNetworkConfig` in several places in the commit path as well. Given the amount of uncopy-able state that we have to pass const references to, I don't think there's a low-hanging way to have stronger invariants for some of this state that needs a const getter in multiple phases

Reply:
can the commit path just use the mutable version instead of const getter (but still use const& for the temp var)? we now have different semantics for each member of ApplyState - for some we only allow const getters in APPLYING phase, but for others we allow const getters in all phases (this is one of the reasons I was asking if we needed the asserts in const getters to begin with).

Reply:
I still think an assert for const getters are useful for state that follows the read-only during apply phase, modify once during commit phase pattern. This applies to module and state caches. Network Config gets a little tricky, since the config needs to be read across both stages. I can add the assert, but forcing the commit phase to always get the mutable reference doesn't seem ideal when they just want a const getter.